service/frontend/workflowHandler.go

```diff
@@ -2213,17 +1884,31 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 				token = nil
 			}
 		} else {
-			history, token.PersistenceToken, err = wh.getHistory(
-				scope,
-				domainID,
-				*execution,
-				token.FirstEventID,
-				token.NextEventID,
-				getRequest.GetMaximumPageSize(),
-				token.PersistenceToken,
-				token.TransientDecision,
-				token.BranchToken,
-			)
+			if rawHistoryQueryEnabled {
+				historyBlob, token.PersistenceToken, err = wh.getRawHistory(
+					scope,
+					domainID,
+					*execution,
+					token.FirstEventID,
+					token.NextEventID,
+					getRequest.GetMaximumPageSize(),
+					token.PersistenceToken,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+			} else {
+				history, token.PersistenceToken, err = wh.getHistory(
+					scope,
+					domainID,
+					*execution,
+					token.FirstEventID,
+					token.NextEventID,
+					getRequest.GetMaximumPageSize(),
+					token.PersistenceToken,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+			}
```

Comment:
consider creating a helper method that takes the enabledFlag as param and returns 3 results - history, rawHistory, error.

Reply:
> consider creating a helper method that takes the enabledFlag as param and returns 3 results - history, rawHistory, error.

Seems not only this flag as argument, also need to take token and getRequest as well.